src/Anexia.MathematicalProgram/Solve/LinearProgramSolver.cs

```diff
@@ -0,0 +1,111 @@
+﻿// ------------------------------------------------------------------------------------------
+//  <copyright file = "LinearProgramSolver.cs" company = "ANEXIA® Internetdienstleistungs GmbH">
+//  Copyright (c) ANEXIA® Internetdienstleistungs GmbH.All rights reserved.
+//  </copyright>
+//  ------------------------------------------------------------------------------------------
+
+#region
+
+using Anexia.MathematicalProgram.Model;
+using Anexia.MathematicalProgram.Result;
+using Anexia.MathematicalProgram.SolverConfiguration;
+using Google.OrTools.LinearSolver;
+
+#endregion
+
+namespace Anexia.MathematicalProgram.Solve;
+
+public sealed class LinearProgramSolver : MemberwiseEquatable<LinearProgramSolver>
+{
+    private LinearProgramSolver(Solver solver, SolverParameter solverParameter)
+    {
+        Solver = solver;
+        SolverParameter = solverParameter;
+    }
+
+    public LinearProgramSolver()
+        : this(Solver.CreateSolver(LPSolverType.GLOP.ToString()), new())
+    { }
+
+    private Solver Solver { get; }
+    private SolverParameter SolverParameter { get; }
+
+    public LinearProgramSolver SetSolverConfigurations(SolverParameter solverParameter)
```

Comment:
The name implies that I set some configuration but the method acts more like a factory maybe there is a better name? Also are we sure that it doesn't change the config of the current solver?